### ISSUE TITLE
Do not store `value` in data attribute of li element

### ISSUE THREAD

**Issue by cmbankester** [immense/immybox#23]
This is more of a feature request, but is also a bug report. Assume we have the following setup:
```javascript
const choices = [
  {text: 'Foo', value: {foo: 'foo'}},
  {text: 'Bar', value: {foo: 'bar'}}
];

$('#input-1').immybox({choices});
```

Then only the first item in the immybox is selectable, since each item's `value` is stored and accessed via the data attribute of the corresponding `li`, which converts to a string as `"[object Object]"`.

We can see this by logging on update:
```javascript
$('#input-1').on('update', (e, new_val) => console.log(new_val));
// When selecting any value:
// => {foo: 'foo'}
```

[See JSBin](https://jsbin.com/xahutomequ/edit?js,output)

I propose we instead store each item's index (i.e., it's index in the choices array) in the data attribute, and look up the item from the choices array when it is needed. That should allow `value`s to be any JS object.

**Comment by cbankester** [comment]
I made some good headway with this, as well as converted the code from CoffeeScript to ES6. It's in the feature/data-array branch

**Comment by cmbankester** [comment]
This has been completed. It's currently merged into the `releases/1.0.0` branch.
